dropwizard/src/main/java/com/linecorp/armeria/server/dropwizard/ArmeriaServerFactory.java

```diff
@@ -0,0 +1,222 @@
+    *
+    * @param accessLogWriter - an instance of a {#link AccessLogWriter}
+    */
+    public void setAccessLogWriter(@Valid final AccessLogWriterFactory accessLogWriter) {
+        this.accessLogWriter = Objects.requireNonNull(accessLogWriter,
+                                          "server[type=\"armeria\"].accessLogWriter");
+    }
+
+    public Size getMaxRequestLength() {
+        return maxRequestLength;
+    }
+
+    public void setMaxRequestLength(final Size maxRequestLength) {
+        this.maxRequestLength = maxRequestLength;
+    }
+
+    @JsonIgnore
+    public ServerBuilder getServerBuilder() {
+        return serverBuilder;
+    }
+
+    @Override
+    public Server build(final Environment environment) {
+        this.printBanner(environment.getName());
+        final MetricRegistry metrics = environment.metrics();
+        final ThreadPool threadPool = this.createThreadPool(metrics);
+        final Server server = this.buildServer(environment.lifecycle(), threadPool);
+
+        final JerseyEnvironment jersey = environment.jersey();
+        if (!isJerseyEnabled()) {
+            jersey.disable();
+        }
+
+        final Handler applicationHandler = this.createAppServlet(server,
+                jersey,
+                environment.getObjectMapper(),
+                environment.getValidator(),
+                environment.getApplicationContext(),
+                environment.getJerseyServletContainer(),
+                metrics);
+        final Handler adminHandler = this.createAdminServlet(server,
+                environment.getAdminContext(),
+                metrics,
+                environment.healthChecks());
+        final ContextRoutingHandler routingHandler = new ContextRoutingHandler(ImmutableMap.of(
+                getApplicationContextPath(), applicationHandler,
+                getAdminContextPath(), adminHandler));
+        final Handler gzipHandler = this.buildGzipHandler(routingHandler);
+        server.setHandler(this.addStatsHandler(this.addRequestLog(server, gzipHandler, environment.getName())));
+
+        setupArmeria(server, metrics);
+
+        return server;
+    }
+
+    private void setupArmeria(final Server server, final MetricRegistry metricRegistry) {
+        LOGGER.debug("Building Armeria Server");
+        final JettyService jettyService = JettyService.forServer(server);
+        this.serverBuilder = com.linecorp.armeria.server.Server.builder()
+                .meterRegistry(DropwizardMeterRegistries.newRegistry(metricRegistry))
+                .serviceUnder("/", jettyService);
+
+        if (this.connector != null) {
+            try {
+                setupAmeriaHttp(this.serverBuilder, this.connector);
+            } catch (SSLException | CertificateException e) {
+                LOGGER.error("Unable to define TLS Server", e);
+            }
+        }
+
+        // TODO: Add more items to server builder via Configuration
```

Comment:
Would appreciate a comment on what else should be added to the builder here based on values available at https://www.dropwizard.io/en/stable/manual/configuration.html#servers